File: Kritika Tripathi/Basics Of Programming/Patterns/Pattern20.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cin >> n;
    int scale = n/2+1;
    //write your code here
    for(int i=1; i<=n; i++){
        for(int j=1; j<=n; j++){
            if(j==1 || j==n){
                cout<<"*\t";
            }
            else if(i>=scale){
                if((i+j)==n+1 || (j==i)){
                    cout<<"*\t";
                }
                else{
                    cout<<"\t";
                }
            }
            else{
                cout<<"\t";
            }
        }
        cout<<endl;
    }

}

File: V.Vennela/Basics of Programming/Patterns/Pattern20.java
import java.util.*;

public class Pattern20 {

    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        // write ur code here
        int n = scn.nextInt();
        int x = n/2+1;
        for(int i=1;i<=n;i++)
        {
            for(int j=1;j<=n;j++)
            {
                int a=0;
                if(j>x)
                {
                    a=n%j+1;
                }
                else
                {
                    a=j;
                }
                if(a==1)
                     System.out.print("*\t");
                else if(i+a==n+1)
                {
                    System.out.print("*\t");
                }
                else
                {
                     System.out.print("\t");
                }
                
            }
            System.out.println();
        }

    }
}
